app/src/main/java/com/internship/move/feature/onboarding/OnboardingFragment.kt

```diff
@@ -15,12 +15,72 @@ class OnboardingFragment : Fragment(R.layout.fragment_onboarding) {
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
-        initListeners()
+        initViewPager()
+        initNextButton()
     }
 
-    private fun initListeners() {
-        binding.onboardingBtn.setOnClickListener {
-            findNavController().navigate(OnboardingFragmentDirections.actionGlobalRegisterFragment())
+    private fun initViewPager() {
+        binding.onboardingVP2.isUserInputEnabled = false
+        binding.onboardingVP2.adapter = OnboardingItemAdapter(this, initList())
+        binding.dotsIndicator.attachTo(binding.onboardingVP2)
+    }
+
+    private fun initNextButton() {
+        binding.nextItemBtn.setOnClickListener {
+            if (binding.onboardingVP2.currentItem < binding.onboardingVP2.adapter?.itemCount?.minus(1)!!) {
```

Comment:
Since you're calling adapter methods in multiple places outside the init method, you should store it as a field in the fragment, and use that variable for calls and remove the need for null checks.
Or, store the list as a field and use `lastIndex` or `size` as necessary